clusterspec/clusterspec_test.go

```diff
@@ -0,0 +1,72 @@
+package clusterspec
+
+import (
+	"reflect"
+	"testing"
+
+	v1alpha1core "github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	v1alpha1provider "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+)
+
+func Test_ClusterSpec_Factory_Construction(t *testing.T) {
+	testCases := []struct {
+		description  string
+		config       *Config
+		errorMatcher func(error) bool
+	}{}
```

Comment:
No test cases before we agree on general design of implementation. These tests will only be to guard rules on implementation here.